src/CSnakes.Runtime/Python/GIL.cs

```diff
@@ -83,7 +83,20 @@ public void Dispose()
         public int RecursionCount => recursionCount;
     }
 
-    public static IDisposable Acquire()
+#pragma warning disable IDE0250 // Make struct 'readonly' (logically read-write)
```

Comment:
Not trying to start a big debate here, but this type isn't read-write. A field of it is being modified but not the field itself, so the type should be marked `readonly`. I'm not sure if the C# language team has chimed in on this and likely should, but for now I would marked this as `readonly` also.

Note that if `PyGilState` was a value type, I would agree with the above warning suppresion.

Reply:
> I'm not sure if the C# language team has chimed in on this and likely should, but for now I would marked this as `readonly` also.

I think the way they've chimed in on this is that `readonly` doesn't mean _pure_:

> It is worth noting that _readonly instance member_ != _pure instance member_. A _pure_ instance member guarantees no state will be modified. A `readonly` instance member only guarantees that instance state will not be modified.

So fair call and added the `readonly` modifier with 8c6a4bb82da571f2cf7c23bdf7e90438898a8c3d.